### PR TITLE
♻️ Code cleaning for amp-story bookend

### PR DESCRIPTION
**Pull request by Enriqe**
Revisiting #14613 to do some code cleaning.

### CODE REVIEW

**Comment by Enriqe** [comment]
Initial revision. Still need to do some more for compiler to be happy.

**Comment by Enriqe** [comment]
Ready to merge! ⛴

File: extensions/amp-story/1.0/bookend/components/article.js [hunk 1]
```diff
@@ -35,27 +33,24 @@ export let BookendArticleComponentDef;
...
-
+export class ArticleComponent {
   /** @override */
-  static isValid(articleJson) {
+  isValid(articleJson) {
     if (!articleJson['title'] || !articleJson['url']) {
-      user().error(TAG,
-          'Articles must contain `title` and `url` fields, skipping invalid.');
-      return false;
+      return user().assert(false, 'Articles must contain `title` and `url` ' +
```
Comment (newmuis) [inline extensions/amp-story/1.0/bookend/components/article.js]: Sorry, I'm confused by the `user().assert(false, ...)` paradigm here.  Isn't this functionally different from the `user.error(...); return false;` paradigm used in the old version?

I would have expected the old version to report an error when this is invalid, then return false.
I would expect the new version to report this error and throw an exception when this is invalid.

I think we should pick which paradigm is the desired one, but if it's this, then we should change the `isValid` name to something like `assertValidity` and remove its return value.

File: extensions/amp-story/1.0/bookend/bookend-component.js [hunk 2]
```diff
@@ -30,14 +30,29 @@ export let BookendDataDef;
...
+ * @return {?BookendComponentClass}
  */
-export const componentsMap = {
-  'small': ArticleComponent,
-  'article-set-title': ArticleTitle,
-};
+function componentClassFor(componentType) {
+  switch (componentType) {
+    case 'small':
+      return articleComponent;
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/bookend/bookend-component.js]: This code is very confusing for two reasons:

- the function is named ``componentClassFor`` when it actually returns an instance, and not a class
- the ``articleComponent`` name really makes it feel like it's going to be an instance of a component, where one could expect to find properties (the component data) etc, when it's actually a singleton that behaves as a builder

What do you think of renaming:

- ``componentClassFor`` into ``componentBuiderInstanceFor``
- ``articleComponent`` into ``articleComponentBuilder``

**Pull request merged**